### PR TITLE
feat(jsonnet): allow jsonnet users to specify a lib directory

### PR DESCRIPTION
**Pull request by emjin**
Add `R2C_INTERNAL_JSONNET_LIB` so that you can specify a jsonnet lib directory.

Test plan: make test, see PR

PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [x] Tests included or PR comment includes a reproducible test plan
- [x] Documentation is up-to-date
- [x] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure about any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

**Comment by emjin** [comment]
DO NOT MERGE this until `add-jsonnet-tests` is merged. Also, don't merge this yet because some semantics should be changed

**Comment by aryx** [comment]
maybe add in a comment that jsonnet supports the -J flag for something similar.

**Comment by emjin** [comment]
@underyx I added you to check my test :D

File: cli/tests/e2e/test_check.py
Comment (underyx) [inline cli/tests/e2e/test_check.py]:
```suggestion
    monkeypatch.setenv("R2C_INTERNAL_JSONNET_LIB", "rules/jsonnet/lib")
```

File: cli/tests/e2e/rules/jsonnet/python/basic.jsonnet
Comment (aryx) [inline cli/tests/e2e/rules/jsonnet/python/basic.jsonnet:1]: nice!

**Review by aryx — APPROVED** [review]

**Pull request merged**
